File: logs_analyzer/app/scala/src/main/scala/com/databricks/apps/logs/TemplateProvider.scala
package com.databricks.apps.logs

import java.io.IOException

import scala.io.Source

/** Provides HTML template for Renderer. */
object TemplateProvider {

  /**
    * Read content of a file as a single string. The file is looked as a resource in classpath.
    *
    * @param fileName File name.
    * @return File content as a string.
    * @throws IOException Failure during file read.
    */
  def fromResource(fileName: String): String = {
    import resource._
    val res = managed(getClass.getClassLoader.getResourceAsStream(fileName))
      .map {
        inputStream =>
          if (inputStream == null)
            throw new IOException("File not found: " + fileName)
          val template = Source.fromInputStream(inputStream).mkString
          template
      }.either.either
    res match {
      case Left(throwables) => throw throwables.head
      case Right(string) => string
    }
  }
}

File: logs_analyzer/app/java8/src/main/java/com/databricks/apps/logs/TemplateProvider.java
package com.databricks.apps.logs;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.Charset;

import org.apache.commons.io.IOUtils;

/**
 * Provides HTML template for Renderer.
 */
public class TemplateProvider {

  public static String fromResource(String fileName) throws IOException {
    try (InputStream input = TemplateProvider.class.getClassLoader().getResourceAsStream(fileName)) {
      if (input == null) {
        throw new IOException("File not found: " + fileName);
      }
      return IOUtils.toString(input, Charset.forName("UTF-8"));
     }
  }
}
